### PR TITLE
Clamp BaseMaterial3D triplanar sharpness to values that never look broken

### LINKED ISSUE #58587: Triplanar sharpness > 180 causes visual glitch
### Godot version

v4.0.alpha3.official

### System information

Windows 10, Vulkan, Realtek Audio

### Issue description

In a StandardMaterial3D resource, setting triplanar sharpness to anything past 180 degrees causes weird visual glitches - it should be clamped to 180 degrees because it doesn't really make sense for the sharpness to ever be > 180 degrees.
Normal triplanar value:
image
Above 180:
image

### Steps to reproduce

Create a MeshInstance3D. Add a StandardMaterial3D to its Surface Material Override. Go to UV1 in the material and enable the Triplanar checkbox. Set triplanar sharpness to anything more than 180 and observe the mesh. Should look distorted.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by Calinou**
`master` version of https://github.com/godotengine/godot/pull/58601.

An alternative solution I was thinking about was to switch to a "hard" cutoff that only performs one sample per pixel when the sharpness is greater than 150. This would also improve performance in situations where you don't need triplanar mapping to smoothly blend between textures (e.g. in hard surface level design with CSG nodes).
That said, this may be better suited to a dedicated boolean property.

This closes https://github.com/godotengine/godot/issues/58587.

**Testing project:** [test_triplanar2.zip](https://github.com/godotengine/godot/files/8149211/test_triplanar2.zip)

### CODE REVIEW

**Comment by pwnorbitals** [comment]
Just wondering for triage, is there anything wrong with this PR ? It is old so we may want to merge it if it's fine :)

**Comment by Calinou** [comment]
> It is old so we may want to merge it if it's fine :)

We don't automatically merge PRs if they're old and have no reviews. We still need review from a qualified maintainer before we can merge this PR.

**Review by clayjohn — APPROVED** [review]

**Pull request merged**
